client/src/analysis/components/Clustering.tsx

```diff
@@ -0,0 +1,212 @@
+        <>{frameViewTitle} Ring: center=(x={cx.toFixed(2)}, y={cy.toFixed(2)}), ri={ri.toFixed(2)}, ro={ro.toFixed(2)}</>
+    )
+    const toolbar = <Toolbar analysis={analysis} onApply={runAnalysis} busyIdxs={[2]} />
+    
+    const [paramsVisible, setParamsVisible] = React.useState(false);
+
+    const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
+        setParamsVisible(!paramsVisible);
+    }
+    
+    const clustparams =
+    <Accordion>
+    <Accordion.Title active={paramsVisible} index={0} onClick={handleClick}>
+          <Icon name='dropdown' />
+          Parameters
+    </Accordion.Title>
+    <Accordion.Content active={paramsVisible}>
+    <Form>
+        <Form.Field> 
+                <label> Delta. Relative intensity difference for decision making for feature vector value (delta = (x-ref)/ref, so, normally, value should be in range [0,1]) </label><input type="number" value={delta} step="0.01" min="0" max="2" onChange={deltaChange}/> 
+        </Form.Field>
+        <Form.Field>
+            <label> Number of clusters </label> <input type="number" value={n_clust}  step="1" min="2" max="100" onChange={clustChange}/> 
+        </Form.Field>    
+        <Form.Field>
+            <label>  Maximal number of possible peak positions to detect (better put higher value,
+        the output is limited to the number of peaks the algorithm could find) </label> <input type="number" value={n_peaks}  step="1" min="5" max="200" onChange={peakChange}/>    
+        </Form.Field>
+        <Form.Field>
+            <label>  Minimal distance between peaks </label> <input type="number" value={min_dist}  step="1" min="0" max="100" onChange={min_distChange}/>    
+        </Form.Field>
+    </Form>
+    </Accordion.Content>
+    </Accordion>
+    return (
+        <AnalysisLayoutTwoRes
+            title="Region clustering" subtitle={subtitle}
+            left={<>
+                <ResultList
+                    extraHandles={frameViewHandles} extraWidgets={frameViewWidgets}
+                    jobIndex={0} analysis={analysis.id}
+                    width={imageWidth} height={imageHeight}
+                    selectors={frameModeSelector}
+                />
+            </>}
+            mid={<>
+                <ResultList
+                    jobIndex={1} analysis={analysis.id}
+                    width={scanWidth} height={scanHeight}
+                    extraHandles={RectRoiHandles}
+                    extraWidgets={RectRoiWidgets}
+                />
+            </>}
+
+            right={<>
+                <ResultList
+                    jobIndex={2} analysis={analysis.id}
+                    width={scanWidth} height={scanHeight}
+                    extraHandles={resultHandles}
+                    extraWidgets={resultWidgets}
+                />
+            </>}
+            toolbar={toolbar}
+            clustparams= {clustparams}
+
+            title1="Frame view. Choose ring parameters for peak finding"
```

Comment:
`title1`: should describe that only peaks inside the ring are included in the feature vector